consul/src/main/java/com/linecorp/armeria/internal/consul/CatalogClient.java

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.internal.consul;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+import javax.annotation.Nullable;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.type.CollectionType;
+import com.fasterxml.jackson.databind.type.TypeFactory;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Strings;
+
+import com.linecorp.armeria.client.Endpoint;
+import com.linecorp.armeria.common.QueryParams;
+
+/**
+ * Consul's catalog API client to registers a service, to deregisters a service and to gets list of services.
+ * {@code CatalogClient} is responsible for endpoint of Consul API:
+ * {@code `/catalog`}(https://www.consul.io/api/catalog.html)
+ *
+ * <p>Consul provides below catalog APIs, but we build interface not for all in this time.
+ * <li> PUT /catalog/register - Register Entity(nodes, services and checks) </li>
+ * <li> PUT /catalog/deregister - Deregister Entity </li>
+ * <li> GET /catalog/datacenters - List Datacenters </li>
+ * <li> GET /catalog/nodes - List Nodes </li>
+ * <li> GET /catalog/services - List Services </li>
+ * <li> GET /catalog/service/:service - List Nodes for Service * </li>
+ * <li> GET /catalog/connect/:service - List Nodes for Connect-capable Service * </li>
+ * <li> GET /catalog/node/:node - List Services for Node </li>
```

Comment:
Please clean up APIs that have been removed.

Reply:
Yes.